packages/components/src/components/textarea/textarea.test.stories.ts

```diff
@@ -0,0 +1,461 @@
+    { type: 'attribute', name: 'help-text', value: 'help-text' }
+  ]),
+  render: (args: any) => {
+    return generateTemplate({
+      axis: {
+        y: { type: 'attribute', name: 'size' }
+      },
+      args
+    });
+  }
+};
+
+/**
+ * Per default the input will indicate an error state when the input is invalid. Use the `style-on-valid` attribute to opt-in the automatic indication of a valid state as well.
+ */
+
+export const StyleOnValid = {
+  name: 'Style on valid',
+  parameters: {
+    controls: {
+      exclude: ['style-on-valid']
+    }
+  },
+  args: overrideArgs([
+    { type: 'attribute', name: 'value', value: 'valu' },
+    { type: 'attribute', name: 'label', value: 'Label' },
+    { type: 'attribute', name: 'help-text', value: 'help-text' },
+    { type: 'attribute', name: 'clearable', value: true },
+    {
+      type: 'slot',
+      name: 'right',
+      value: '<sd-icon slot="right" library="global-resources" name="system/picture"></sd-icon>'
+    }
+  ]),
+  render: (args: any) => {
+    return generateTemplate({
+      axis: {
+        y: { type: 'attribute', name: 'style-on-valid' }
+      },
+      args
+    });
+  },
+
+  play: async ({ canvasElement }: { canvasElement: HTMLUnknownElement }) => {
+    const els = canvasElement.querySelectorAll('sd-textarea');
+
+    for (const el of els) {
+      await waitUntil(() => el?.shadowRoot?.querySelector('textarea'));
+      await userEvent.type(el.shadowRoot!.querySelector('textarea')!, 'e');
+    }
+
+    // tab to next element to loose focus
+    await userEvent.tab();
+  }
+};
+
+/**
+ * Demonstrates the various validation options extended from the native textarea element in addition to error and success styles.
+ */
+
+export const Validation = {
+  name: 'Validation',
+  parameters: {
+    controls: {
+      include: ['clearable', 'disabled']
+    }
+  },
+  render: (args: any) => {
+    return html`
+      <form action="" method="get" id="testForm" name="testForm" class="w-[370px]">
+        <div class="mb-2">
+          ${generateTemplate({
+            constants: [
+              { type: 'attribute', name: 'label', value: 'Required' },
+              { type: 'attribute', name: 'name', value: 'required field' },
+              { type: 'attribute', name: 'placeholder', value: '.*' },
+              { type: 'attribute', name: 'help-text', value: 'textarea must be filled' },
+              { type: 'attribute', name: 'form', value: 'testForm' },
+              { type: 'attribute', name: 'required', value: true },
+              { type: 'attribute', name: 'style-on-valid', value: true }
+            ],
+            args
+          })}
+        </div>
+        <div class="mb-2">
+          ${generateTemplate({
+            constants: [
+              { type: 'attribute', name: 'label', value: 'Min Length' },
+              { type: 'attribute', name: 'name', value: 'min length field' },
+              { type: 'attribute', name: 'placeholder', value: '^.{3,}$' },
```

Comment:
I don’t get the placeholders ^.{3,}$ ... is this useful? If it does not support any information I would delete it.